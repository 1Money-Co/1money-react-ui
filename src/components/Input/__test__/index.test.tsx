import 'jsdom-global/register';
import * as React from 'react';
import { configure, shallow, render, mount } from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import { Input } from '../index';

configure({ adapter: new Adapter() });

const originalConsoleError = console.error;
console.error = (message, ...optionalParams) => {
  if (
    message.includes('Could not parse CSS stylesheet') ||
    message.includes('findDOMNode is deprecated and will be removed') ||
    message.includes('React does not recognize') ||
    message.includes('should not be null') ||
    message.includes('A component is changing an uncontrolled input to be controlled')
  ) {
      return;
  }
  originalConsoleError(message, ...optionalParams);
};

describe('Input', () => {
  it('renders correctly', () => {
    const text = render(
      <Input />
    );
    const autocomplete = shallow(
      <Input type='autocomplete' />
    );
    const mask = mount(
      <Input type='mask' mask='(999) 999-9999' />
    );
    const number = render(
      <Input type='number' />
    );
    const otp = shallow(
      <Input type='otp' />
    );
    const password = mount(
      <Input type='password' />
    );
    const textarea = render(
      <Input type='textarea' />
    );
    expect(text).toMatchSnapshot();
    expect(autocomplete).toMatchSnapshot();
    expect(mask).toMatchSnapshot();
    expect(number).toMatchSnapshot();
    expect(otp).toMatchSnapshot();
    expect(password).toMatchSnapshot();
    expect(textarea).toMatchSnapshot();
  });
});
