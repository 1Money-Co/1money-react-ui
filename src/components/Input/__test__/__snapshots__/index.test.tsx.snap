// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
LoadedCheerio {
  "0": <div
    class="usd1-react-ui-input-wrapper"
  >
    <input
      class="usd1-react-ui-input usd1-react-ui-input-text usd1-react-ui-input-rounded  p-inputtext p-component"
      data-pc-name="inputtext"
      data-pc-section="root"
    />
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Input renders correctly 2`] = `ShallowWrapper {}`;

exports[`Input renders correctly 3`] = `ReactWrapper {}`;

exports[`Input renders correctly 4`] = `
LoadedCheerio {
  "0": <div
    class="usd1-react-ui-input-wrapper"
  >
    <span
      class="usd1-react-ui-input usd1-react-ui-input-number usd1-react-ui-input-rounded  p-inputnumber p-component p-inputwrapper"
      data-pc-name="inputnumber"
      data-pc-section="root"
    >
      <input
        class="p-inputnumber-input p-inputtext p-component"
        data-pc-name="inputtext"
        data-pc-section="root"
        inputmode="numeric"
        role="spinbutton"
        type="text"
        value=""
      />
    </span>
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Input renders correctly 5`] = `ShallowWrapper {}`;

exports[`Input renders correctly 6`] = `ReactWrapper {}`;

exports[`Input renders correctly 7`] = `
LoadedCheerio {
  "0": <div
    class="usd1-react-ui-input-wrapper"
  >
    <textarea
      class="usd1-react-ui-input usd1-react-ui-input-textarea usd1-react-ui-input-rounded  p-inputtextarea p-inputtext p-component usd1-react-ui-input usd1-react-ui-input-textarea usd1-react-ui-input-rounded"
      data-pc-name="inputtextarea"
      data-pc-section="root"
    />
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
